[9.0][FIX]product_variant_configurator: fix create_variant_ids

zakiuu opened the pull request:
In standard, when attributes are added to an already created product
template, the variant without attributes is deleted or deactivated (been
used somewhere else) automatically.
In product variant configurator module we loosing that behavior when the
creation of the variants are delayed (i.e case where no auto variants are created).
This PR aims to restore that behavior by deactivating the variant with
no attributes.

zakmak88 committed:
[9.0][FIX]product_variant_configurator: fix create_variant_ids

In standard, when attributes are added to an already created product
template, the variant without attributes is deleted or deactivated (been
used somewhere else) automatically.
In product variant configurator module we loosing that behavior when the
creation of the variants are delayed (i.e case where no auto variants are created).
This PR aims to restore that behavior by deactivating the variant with
no attributes.

andreparames commented:
LGTM :+1:

pedrobaeza reviewed (changes requested):
If you change this, then the warning on the onchange of the field should be updated for reflecting this.

zakmak88 committed:
[9.0][REF]product_variant_configurator: refactor onchange_no_create_variants warning msg

zakiuu commented:
@pedrobaeza thanks for your review, I've changed. Hope it looks clear now

pedrobaeza commented:
I'm afraid if doesn't serve, as you have change the message inside the `if` clause. Please put 2 different messages depending on the value.

pedrobaeza commented:
Are you going to continue this?

ThomasBinsfeld commented:
@pedrobaeza I would like to finish this one. Could you explain me what must be done?

ThomasBinsfeld committed:
[REF] Product Template: onchange no create variant warning
The warning on no_create_variant must be triggered whatever the value (yes, no, empty)
